auto_submit/lib/request_handling/authentication.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:meta/meta.dart';
+import 'package:shelf/shelf.dart';
+
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../requests/exceptions.dart';
+import '../service/config.dart';
+
+/// Class capable of authenticating [HttpRequest]s.
+///
+///  If the request has the `'X-Appengine-Cron'` HTTP header set to "true",
+///  then the request will be authenticated as an App Engine cron job.
+///
+///  The `'X-Appengine-Cron'` HTTP header is set automatically by App Engine
+///  and will be automatically stripped from the request by the App Engine
+///  runtime if the request originated from anything other than a cron job.
+///  Thus, the header is safe to trust as an authentication indicator.
+///
+///
+/// If none of the above authentication methods yield an authenticated
+/// request, then the request is unauthenticated, and any call to
+/// [authenticate] will throw an [Unauthenticated] exception.
+///
+/// See also:
+///
+///  * <https://cloud.google.com/appengine/docs/standard/python/reference/request-response-headers>
+@immutable
+class AuthenticationProvider {
```

Comment:
Since this isn't generic authentication, you should call it something like `CronAuthProvider`

Reply:
Sure, changed the class name to CronAuthProvider.